plugin/src/main/java/org/autorefactor/refactoring/rules/AndroidViewHolderRefactoring.java

```diff
@@ -0,0 +1,320 @@
+                                    b.name("viewHolderItem"), b.simpleName(item.variable.getIdentifier()));
+                            r.replace(item.findViewByIdExpression, b.copy(qualifiedName));
+                        }
+                        // store viewHolderItem in convertView
+                        MethodInvocation setTagInvocation = b.invoke("convertView", "setTag",
+                                b.simpleName("viewHolderItem"));
+                        thenBlock.statements().add(b.getAST().newExpressionStatement(setTagInvocation));
+
+                        // retrieve viewHolderItem from convertView
+                        ifStatement.setElseStatement(b.block(b.getAST().newExpressionStatement(
+                                b.assign(b.simpleName("viewHolderItem"), Assignment.Operator.ASSIGN,
+                                        b.cast("ViewHolderItem", b.invoke("convertView", "getTag"))))));
+
+                    }
+                    r.remove(visitor.viewAssignmentStatement);
+                    return DO_NOT_VISIT_SUBTREE;
+                }
+            }
+        }
+        return VISIT_SUBTREE;
+    }
+
+    private static boolean isInflateMethod(MethodInvocation node) {
+        return isMethod(node, "android.view.LayoutInflater", "inflate", "int", "android.view.ViewGroup")
+                || isMethod(node, "android.view.LayoutInflater", "inflate", "int", "android.view.ViewGroup", "boolean")
+                || isMethod(node, "android.view.LayoutInflater", "inflate", "org.xmlpull.v1.XmlPullParser",
+                        "android.view.ViewGroup")
+                || isMethod(node, "android.view.LayoutInflater", "inflate", "org.xmlpull.v1.XmlPullParser",
+                        "android.view.ViewGroup", "boolean");
+    }
+
+    private class GetViewVisitor extends ASTVisitor {
+        private SimpleName viewVariable = null;
+        private Statement viewAssignmentStatement;
+        private VariableDeclarationFragment viewVariableDeclarationFragment = null;
+        private Assignment viewVariableAssignment = null;
+        private ReturnStatement returnStatement = null;
+
+        GetViewVisitor() {
+        }
+
+        @Override
+        public boolean visit(MethodInvocation node) {
+            if (isInflateMethod(node)) {
+                this.viewVariableDeclarationFragment = (VariableDeclarationFragment) getFirstAncestorOrNull(node,
+                        VariableDeclarationFragment.class);
+                if (viewVariableDeclarationFragment != null) {
+                    this.viewVariable = viewVariableDeclarationFragment.getName();
+                    this.viewAssignmentStatement = getAncestor(viewVariableDeclarationFragment,
+                            VariableDeclarationStatement.class);
+                } else {
+                    this.viewVariableAssignment = (Assignment) getFirstAncestorOrNull(node, Assignment.class);
```

Comment:
I'm not comfortable with using ``getAncestor`` because I am not sure whether the result will actually be not null..